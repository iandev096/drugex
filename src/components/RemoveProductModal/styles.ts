import { StyleSheet } from "react-native";
import { Theme } from "../../../theme";
import { HEIGHT, WIDTH } from "../../constants/dimensions";

const styles = (theme: Theme) =>
  StyleSheet.create({
    container: {
      backgroundColor: theme.colors.background,
      padding: 20,
      width: WIDTH * 0.88,
    },
    center: {
      alignItems: "center",
    },
    buttonGroup: {
      flexDirection: "row",
      justifyContent: "space-between",
      marginTop: HEIGHT * 0.05,
    },
    button: {
      flex: 0.47,
    },
    promptText: {
      textAlign: "center",
      marginBottom: 14,
    },
    productName: {
      fontStyle: "italic",
      color: theme.colors.error,
    },
  });

export default styles;
