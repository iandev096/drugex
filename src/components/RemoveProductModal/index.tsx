import { View } from "react-native";
import { Button, Modal, Portal, Text } from "react-native-paper";
import { Product } from "../../contexts/app/type";
import useAppTheme from "../../hooks/useAppTheme";
import styles from "./styles";

export type RemoveProductModalProps = {
  visible: boolean;
  product: Product | null;
  onDismiss?: () => void;
  onRemove?: () => void;
};

export default function RemoveProductModal({
  visible,
  product,
  onDismiss,
  onRemove,
}: RemoveProductModalProps) {
  const theme = useAppTheme();

  return (
    <Portal>
      <Modal
        visible={visible}
        onDismiss={onDismiss}
        contentContainerStyle={styles(theme).container}
        style={styles(theme).center}
      >
        <View style={styles(theme).center}>
          <Text style={styles(theme).promptText}>
            Do you really want to remove this product?
          </Text>
          <Text style={styles(theme).productName}>"{product?.name}"</Text>
        </View>
        <View style={styles(theme).buttonGroup}>
          <Button
            style={styles(theme).button}
            mode="contained"
            icon="close"
            onPress={onDismiss}
            testID='btn:onDismiss'
          >
            Cancel
          </Button>
          <Button
            style={styles(theme).button}
            mode="contained"
            icon="trash-can-outline"
            onPress={onRemove}
            testID='btn:onRemove'
          >
            Remove
          </Button>
        </View>
      </Modal>
    </Portal>
  );
}
