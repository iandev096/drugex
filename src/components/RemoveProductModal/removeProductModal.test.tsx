import { fireEvent, render } from "@testing-library/react-native";
import { Provider } from "react-native-paper";
import RemoveProductModal, { RemoveProductModalProps } from ".";

const props: RemoveProductModalProps = {
    product: { id: 1, name: 'Aracetal 50mg' },
    visible: false,
    onDismiss: () => console.log('onDismiss triggered'),
    onRemove: () => console.log('onRemove triggered')
}

const Component = (props: RemoveProductModalProps) => (
    <Provider>
        <RemoveProductModal {...props} />
    </Provider>
)

test('should render nothing when visible is false', () => {
    const { queryByText } = render(<Component {...props} />);
   
    expect(queryByText(new RegExp(props.product!.name, 'i'))).toBeNull();
});

test('should render modal when visible is true', async () => {
    const { findByText } = render(<Component {...props} visible={true} />);
    
    const productNameRegex = new RegExp(props.product!.name, 'i');
    
    expect((await findByText(productNameRegex))).toBeTruthy();
});


test('renders correctly', () => {
    const { toJSON } = render(<Component {...props} />);

    expect(toJSON()).toMatchSnapshot();
});

test('all buttons work', async () => {
    const { findByTestId, getByTestId } = render(<Component {...props} visible={true} />);
    
    const cancelButton = await findByTestId('btn:onDismiss');
    const deleteButton = getByTestId('btn:onRemove');
    const logSpy = jest.spyOn(console, 'log');
    fireEvent.press(cancelButton);
    fireEvent.press(deleteButton);

    expect(logSpy).toHaveBeenCalledTimes(2);
});


test('renders correctly', () => {
    const { toJSON } = render(<Component {...props} visible={true} />);

    expect(toJSON()).toMatchSnapshot();
});