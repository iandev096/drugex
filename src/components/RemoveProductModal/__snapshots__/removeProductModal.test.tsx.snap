// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;

exports[`renders correctly 2`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;
